src/System.CodeDom/src/System/CodeDom/Compiler/CompilerInfo.cs

```diff
@@ -103,7 +103,7 @@ internal CompilerInfo(CompilerParameters compilerParams, string codeDomProviderT
         public override bool Equals(object o)
         {
             CompilerInfo other = o as CompilerInfo;
-            if (o == null)
+            if (other == null)
```

Comment:
I would be glad to do this, but I doubt that my manager would approve these actions. Most likely he will say something "We are a small company and cannot develop Microsoft projects for free" :) If you wish, I could give you contacts of our CEO and you could discuss variants of paid collaboration. We really could fix all the warnings and write tests, we do such tasks. Here is the example of how we did it for [Unreal Engine](https://www.unrealengine.com/blog/how-pvs-studio-team-improved-unreal-engines-code).